python/cudf/cudf/core/dataframe.py

```diff
@@ -1924,9 +1966,12 @@ def as_gpu_matrix(self, columns=None, order="F"):
         ncol = len(cols)
         nrow = len(self)
         if ncol < 1:
-            raise ValueError("require at least 1 column")
-        if nrow < 1:
-            raise ValueError("require at least 1 row")
+            # This is the case for empty dataframe - construct empty cupy array
+            matrix = cupy.empty(
+                shape=(0, 0), dtype=np.dtype("float64"), order=order
+            )
+            return cuda.as_cuda_array(matrix).view(np.dtype("float64"))
```

Comment:
Why do we need to view it as float64 when the input was already float64?

Reply:
Not needed actually, removed.